src/main/java/io/druid/segment/loading/DataSegmentPuller.java

```diff
@@ -10,10 +13,49 @@
 {
   /**
    * Pull down segment files for the given DataSegment and put them in the given directory.
+   * Deprecated in favor of URI pulling when possible
    *
    * @param segment The segment to pull down files for
-   * @param dir The directory to store the files in
+   * @param dir     The directory to store the files in
+   *
    * @throws SegmentLoadingException if there are any errors
    */
+  @Deprecated
   public void getSegmentFiles(DataSegment segment, File dir) throws SegmentLoadingException;
+
+  /**
+   * Pull down segment files for the given DataSegment and put them in the given directory.
+   *
+   * @param segmentUri The URI for the segment
+   * @param dir        The directory to store the files in
+   *
+   * @throws SegmentLoadingException if there are any errors
+   */
+  public void getSegmentFiles(URI segmentUri, File dir) throws SegmentLoadingException;
```

Comment:
I'm not against making the loader use a unified object, but I don't think Segment is the right choice, and anything that makes a more significant change I worry is scope creep rather than an actual need for the PR to be good.